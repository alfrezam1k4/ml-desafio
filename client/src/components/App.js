import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import { Layout } from '.';

const App = () => {
  return (
    <Router>
      <Route render={() => <Layout />} />
    </Router>
  );
};

export default App;
