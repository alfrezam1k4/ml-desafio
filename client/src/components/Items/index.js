import ItemsResult from './ItemsResult';
import Item from './Item';
import ItemDetails from './ItemDetails';

export { ItemsResult, Item, ItemDetails };
