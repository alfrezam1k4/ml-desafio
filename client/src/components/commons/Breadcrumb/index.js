import React from 'react';
import PropTypes from 'prop-types';
import './Breadcrumb.scss';
const Breadcrumb = ({ categories }) => {
  return (
    categories.length > 0 && (
      <p className="breadcrumb">
        {categories.map((crumb, index) => {
          if (index === categories.length - 1)
            return (
              <span key={index}>
                <b> {crumb}</b>
              </span>
            );
          return <span key={index}> {`${crumb} >`} </span>;
        })}
      </p>
    )
  );
};

Breadcrumb.propTypes = {
  categories: PropTypes.array,
};

export default Breadcrumb;
