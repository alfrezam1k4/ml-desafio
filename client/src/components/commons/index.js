import { Row, Col } from './Grid';
import Breadcrumb from './Breadcrumb';
export { Row, Col, Breadcrumb };
