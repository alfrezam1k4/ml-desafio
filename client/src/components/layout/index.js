import AppBar from './AppBar';
import Layout from './Layout';
import Background from './Background';

export { AppBar, Layout, Background };
