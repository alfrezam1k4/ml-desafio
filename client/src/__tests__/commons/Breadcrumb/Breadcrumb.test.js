import React from 'react';
import renderer from 'react-test-renderer';
import { Breadcrumb } from '../../../components/commons';

const categories = ['1', '3'];
describe('<Breadcrumb />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Breadcrumb categories={categories} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
