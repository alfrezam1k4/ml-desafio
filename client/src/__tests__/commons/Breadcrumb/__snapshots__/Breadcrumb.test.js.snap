// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> renders correctly 1`] = `
<p
  className="breadcrumb"
>
  <span>
     
    1 &gt;
     
  </span>
  <span>
    <b>
       
      3
    </b>
  </span>
</p>
`;
